alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/InternalConnectorRegistry.java

```diff
@@ -55,6 +56,12 @@ enum InternalConnectorRegistry implements Closeable {
     this.credentialFactoryProvider = new CredentialFactoryProvider();
   }
 
+  /** Test use only: Set a new CredentialFactoryProvider */
+  @VisibleForTesting
```

Comment:
Generally, I think we should avoid this annotation and test through the public API. However, since this is an enum, we're in a tight corner and so this is probably the most reasonable solution. If we find ourselves reaching for this annotation again, let's revisit if an enum singleton is serving us well.

Reply:
Agreed